modules/justpremiumBidAdapter.js

```diff
@@ -1,274 +1,186 @@
-import bidfactory from 'src/bidfactory';
-import bidmanager from 'src/bidmanager';
-import adloader from 'src/adloader';
-import * as utils from 'src/utils';
-import adaptermanager from 'src/adaptermanager';
-
-const CONSTANTS = require('src/constants.json');
-
-const JustpremiumAdapter = function JustpremiumAdapter() {
-  const top = window.top;
-  let d;
-  let bids;
-  let cookieLoaded = false;
-  let adManagerLoaded = false;
-  let jPAM;
-  let dConfig;
-  let toLoad;
-  let server;
-
-  function isCrossOriginIframe() {
-    try {
-      return !top.document;
-    } catch (e) {
-      return true;
+import { registerBidder } from 'src/adapters/bidderFactory'
+
+const BIDDER_CODE = 'justpremium'
+const ENDPOINT_URL = top.document.location.protocol + '//pre.ads.justpremium.com/v/2.0/t/xhr'
+const pixels = [{
+  type: 'iframe',
+  src: '//us-u.openx.net/w/1.0/pd?plm=10&ph=26e53f82-d199-49df-9eca-7b350c0f9646'
+}]
+
+export const spec = {
+  code: BIDDER_CODE,
+  time: 60000,
+
+  isBidRequestValid: (bid) => {
+    return !!(bid && bid.params && bid.params.zone)
+  },
+
+  buildRequests: (validBidRequests) => {
+    const c = preparePubCond(validBidRequests)
+    const payload = {
+      zone: [...new Set(validBidRequests.map(b => {
+        return parseInt(b.params.zone)
+      }))].join(','),
+      hostname: top.document.location.hostname,
+      protocol: top.document.location.protocol.replace(':', ''),
+      sw: window.top.screen.width,
+      sh: window.top.screen.height,
+      ww: window.top.innerWidth,
+      wh: window.top.innerHeight,
+      c: c,
+      id: validBidRequests[0].params.zone,
+      sizes: {}
     }
-  }
-
-  function arrayUnique(array) {
-    const a = array.concat();
-    for (let i = 0; i < a.length; ++i) {
-      for (let j = i + 1; j < a.length; ++j) {
-        if (a[i] === a[j]) {
-          a.splice(j--, 1);
+    validBidRequests.forEach(b => {
+      const zone = b.params.zone
+      const sizes = payload.sizes
+      sizes[zone] = sizes[zone] || []
+      sizes[zone].push.apply(sizes[zone], b.sizes)
+    })
+    const payloadString = JSON.stringify(payload)
+
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL + '?i=' + (+new Date()),
+      data: payloadString,
+      bids: validBidRequests
+    }
+  },
+
+  interpretResponse: (serverResponse, bidRequests) => {
+    let bidResponses = []
+    bidRequests.bids.forEach(adUnit => {
+      let bidId = adUnit.bidId
+      let bid = findBid(adUnit.params, serverResponse.bid)
+      if (bid) {
+        let size = (adUnit.sizes && adUnit.sizes.length && adUnit.sizes[0]) || []
+        let bidResponse = {
+          requestId: bidId,
+          bidderCode: spec.code,
+          width: size[0] || bid.width,
+          height: size[1] || bid.height,
+          ad: bid.adm,
+          cpm: bid.price,
+          currency: bid.currency || 'USD',
+          ttl: bid.ttl || spec.time
```

Comment:
`netRevenue` and `creativeId` will also need to be in this object to pass validation, see the table in http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for a description of these fields